cmd/build/v2/services.go

```diff
@@ -74,55 +92,56 @@ func (bc *OktetoBuilder) GetServicesToBuild(ctx context.Context, manifest *model
 	}
 
-	imageWithDigest, err := bc.Registry.GetImageTagWithDigest(tag)
-	if oktetoErrors.IsNotFound(err) {
-		oktetoLog.Debug("image not found, building image")
-		ch <- service
-		return nil
-	} else if err != nil {
-		return fmt.Errorf("error checking image at registry %s: %v", tag, err)
+	for _, tag := range tags {
+		imageWithDigest, err := bc.Registry.GetImageTagWithDigest(tag)
+		if oktetoErrors.IsNotFound(err) {
+			oktetoLog.Debugf("image not found for tag %s, building image", tag)
+			continue
+		} else if err != nil {
+			return "", fmt.Errorf("error checking image at registry %s: %w", tag, err)
+		}
+		return imageWithDigest, nil
 	}
-	oktetoLog.Debug("Skipping build for image for service")
 
-	bc.SetServiceEnvVars(service, imageWithDigest)
+	return "", nil
+}
 
-	if manifest.Deploy != nil && manifest.Deploy.ComposeSection != nil && manifest.Deploy.ComposeSection.Stack != nil {
-		stack := manifest.Deploy.ComposeSection.Stack
-		if svc, ok := stack.Services[service]; ok && svc.Image == "" {
-			stack.Services[service].Image = fmt.Sprintf("${OKTETO_BUILD_%s_IMAGE}", strings.ToUpper(strings.ReplaceAll(service, "-", "_")))
+func getToBuildTags(manifestName, svcName string, b *model.BuildInfo) []string {
+	if !okteto.IsOkteto() {
+		if b.Image != "" {
+			return []string{b.Image}
 		}
+		return []string{}
 	}
-	return nil
-}
 
-func getToBuildTag(manifestName, svcName string, b *model.BuildInfo) string {
-	targetRegistry := okteto.DevRegistry
+	if registry.IsOktetoRegistry(b.Image) {
+		return []string{b.Image}
+	}
+
+	possibleTags := []string{}
 	// manifestName can be not sanitized when option name is used at deploy
 	sanitizedName := format.ResourceK8sMetaString(manifestName)
-	switch {
-	case !okteto.IsOkteto():
-		return b.Image
-	case (shouldBuildFromDockerfile(b) && shouldAddVolumeMounts(b)) || shouldAddVolumeMounts(b):
-		return fmt.Sprintf("%s/%s-%s:%s", targetRegistry, sanitizedName, svcName, model.OktetoImageTagWithVolumes)
-	case b.Image != "" && shouldBuildFromDockerfile(b):
-		return b.Image
-	case shouldBuildFromDockerfile(b):
-		return fmt.Sprintf("%s/%s-%s:%s", targetRegistry, sanitizedName, svcName, model.OktetoDefaultImageTag)
-	case b.Image != "":
-		return b.Image
-	default:
-		oktetoLog.Infof("could not build service %s, due to not having Dockerfile defined or volumes to include", svcName)
+
+	targetRegistries := []string{okteto.DevRegistry, okteto.GlobalRegistry}
+	for _, targetRegistry := range targetRegistries {
+		if shouldAddVolumeMounts(b) {
+			possibleTags = append(possibleTags, fmt.Sprintf("%s/%s-%s:%s", targetRegistry, sanitizedName, svcName, model.OktetoImageTagWithVolumes))
+			continue
+		}
+		if shouldBuildFromDockerfile(b) {
+			possibleTags = append(possibleTags, fmt.Sprintf("%s/%s-%s:%s", targetRegistry, sanitizedName, svcName, model.OktetoDefaultImageTag))
+		}
+	}
```

Comment:
```suggestion
if shouldAddVolumeMounts(b) {
		possibleTags = append(possibleTags, fmt.Sprintf("%s/%s-%s:%s", okteto.DevRegistry, sanitizedName, svcName, model.OktetoImageTagWithVolumes))
		possibleTags = append(possibleTags, fmt.Sprintf("%s/%s-%s:%s", okteto.GlobalRegistry, sanitizedName, svcName, model.OktetoImageTagWithVolumes))
}
if shouldBuildFromDockerfile(b) {
		possibleTags = append(possibleTags, fmt.Sprintf("%s/%s-%s:%s", okteto.DevRegistry, sanitizedName, svcName, model.OktetoDefaultImageTag))
		possibleTags = append(possibleTags, fmt.Sprintf("%s/%s-%s:%s", okteto.GlobalRegistry, sanitizedName, svcName, model.OktetoDefaultImageTag))
	}
```
WDYT? to not check twice `shouldAddVolumeMounts(b)` and `shouldBuildFromDockerfile(b)`. If it's true for `okteto.dev` is also true for `okteto.global`

Reply:
true, this suggestion reduces the complexity of the code. Thanks for spotting it!! 🚀